homeassistant/components/hvac/demo.py

```diff
@@ -0,0 +1,155 @@
+"""
+Demo platform that offers a fake hvac.
+
+For more details about this platform, please refer to the documentation
+https://home-assistant.io/components/demo/
+"""
+from homeassistant.components.hvac import HvacDevice
+from homeassistant.const import TEMP_CELSIUS, TEMP_FAHRENHEIT
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the Demo hvacs."""
+    add_devices([
+        DemoHvac("HeatPump", 68, TEMP_FAHRENHEIT, None, 77, "Auto Low",
+                 None, None, "Auto", "Heat", None),
+        DemoHvac("Hvac", 21, TEMP_CELSIUS, True, 22, "On High",
+                 67, 54, "Off", "Cool", False),
+    ])
+
+
+# pylint: disable=too-many-arguments, too-many-public-methods
+class DemoHvac(HvacDevice):
+    """Representation of a demo hvac."""
+
+    # pylint: disable=too-many-instance-attributes
+    def __init__(self, name, target_temperature, unit_of_measurement,
+                 away, current_temperature, current_fan_mode,
+                 target_humidity, current_humidity, current_swing_mode,
+                 current_operation, aux):
+        """Initialize the hvac."""
+        self._name = name
+        self._target_temperature = target_temperature
+        self._target_humidity = target_humidity
+        self._unit_of_measurement = unit_of_measurement
+        self._away = away
+        self._current_temperature = current_temperature
+        self._current_humidity = current_humidity
+        self._current_fan_mode = current_fan_mode
+        self._current_operation = current_operation
+        self._aux = aux
+        self._current_swing_mode = current_swing_mode
+        self._fan_list = ["On Low", "On High", "Auto Low", "Auto High", "Off"]
+        self._operation_list = ["Heat", "Cool", "Auto Changeover", "Off"]
+        self._swing_list = ["Auto", 1, 2, 3, "Off"]
+
+    @property
+    def should_poll(self):
+        """Polling needed for a demo hvac."""
+        return True
```

Comment:
Setting should_poll to false in the demo makes the test fail, obviously
because in init.py there is
`if hvac.should_poll:
    hvac.update_ha_state(True)`
Did I misunderstand something here?

On Thu, Apr 28, 2016 at 1:08 PM, Paulus Schoutsen <notifications@github.com>
wrote:

> In homeassistant/components/hvac/demo.py
> <https://github.com/home-assistant/home-assistant/pull/1912#discussion_r61409603>
> :
>
> > +        self._unit_of_measurement = unit_of_measurement
> > +        self._away = away
> > +        self._current_temperature = current_temperature
> > +        self._current_humidity = current_humidity
> > +        self._current_fan_mode = current_fan_mode
> > +        self._current_operation = current_operation
> > +        self._aux = aux
> > +        self._current_swing_mode = current_swing_mode
> > +        self._fan_list = ["On Low", "On High", "Auto Low", "Auto High", "Off"]
> > +        self._operation_list = ["Heat", "Cool", "Auto Changeover", "Off"]
> > +        self._swing_list = ["Auto", 1, 2, 3, "Off"]
> > +
> > +    @property
> > +    def should_poll(self):
> > +        """Polling needed for a demo hvac."""
> > +        return True
>
> Polling should not be needed as the current state is always represented on
> the object. Just make sure to add calls to update_ha_state() at the end
> of each service method (set_temperature etc)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/home-assistant/home-assistant/pull/1912/files/1cf74f9d35b0af6bff054b9db17c7e3e22b45a3c#r61409603>
>